app/controllers/samples_controller.rb

```diff
@@ -208,7 +208,7 @@ def bulk_process_csv
       CSV.open(csv_file.path) do |csv_stream|
         csv_stream.each do |row|
           sample_id, measured_signal = row[0], row[1]
-          next if !sample_id.match(uuid_regex) # non uuids are ignored
+          next if !sample_id || !sample_id.match(uuid_regex) # non uuids are ignored
```

Comment:
FYI, the following should do (`&.` is the `lonely operator`, similar to `?.` in JS):

```suggestion
          next unless sample_id&.match(uuid_regex) # non uuids are ignored
```

And below could be (Crystal idiom):

```ruby
if sample = Sample.find_by_uuid(sample_id)
  # ...
end
```

In Ruby (and Crystal) only `nil` and `false` can ever be falsy values, everything else is truthy (even `0`, `""`, ...), so we usually don't have to explicitly test for `.nil?`. In Crystal terms this is a nilable (`Sample | Nil` union) and a simple `if` is enough to skip nils!